radix-engine-interface/src/constants.rs

```diff
@@ -1,67 +1,429 @@
-pub const GLOBAL_OBJECT_TOKEN: ResourceAddress =
-    resource_address(EntityType::GlobalNonFungibleResource, 4);
-pub const PACKAGE_OWNER_TOKEN: ResourceAddress =
-    resource_address(EntityType::GlobalNonFungibleResource, 5);
-pub const VALIDATOR_OWNER_TOKEN: ResourceAddress =
-    resource_address(EntityType::GlobalNonFungibleResource, 6);
-pub const IDENTITY_OWNER_TOKEN: ResourceAddress =
-    resource_address(EntityType::GlobalNonFungibleResource, 7);
-pub const ACCOUNT_OWNER_TOKEN: ResourceAddress =
-    resource_address(EntityType::GlobalNonFungibleResource, 8);
-
-pub const PACKAGE_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 0);
-pub const RESOURCE_MANAGER_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 1);
-pub const IDENTITY_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 2);
-pub const EPOCH_MANAGER_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 3);
-pub const CLOCK_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 4);
-pub const ACCOUNT_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 5);
-pub const ACCESS_CONTROLLER_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 6);
-pub const TRANSACTION_PROCESSOR_PACKAGE: PackageAddress =
-    package_address(EntityType::GlobalPackage, 7);
-pub const METADATA_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 10);
-pub const ROYALTY_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 11);
-pub const ACCESS_RULES_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 12);
-pub const GENESIS_HELPER_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 13);
-pub const FAUCET_PACKAGE: PackageAddress = package_address(EntityType::GlobalPackage, 64);
-
-pub const CLOCK: ComponentAddress = component_address(EntityType::GlobalClock, 0);
-pub const EPOCH_MANAGER: ComponentAddress = component_address(EntityType::GlobalEpochManager, 0);
-pub const FAUCET: ComponentAddress = component_address(EntityType::GlobalGenericComponent, 0);
-pub const GENESIS_HELPER: ComponentAddress =
-    component_address(EntityType::GlobalGenericComponent, 1);
+//=========================================================================
+// Please see and update REP-71 along with changes to this file
+//=========================================================================
 
+//=========================================================================
+// FUNGIBLES
+//=========================================================================
+
+/// XRD is the native token of the Radix ledger.
+/// It is a fungible token, measured in attos (`10^-18`).
+///
+/// It is used for paying fees and staking.
+pub const XRD: ResourceAddress = ResourceAddress::new_or_panic([
+    93, 166, 99, 24, 198, 49, 140, 97, 245, 166, 27, 76, 99, 24, 198, 49, 140, 247, 148, 170, 141,
```

Comment:
I’m wondering if it would be possible to use somehow the values from `EntityType` enum instead of hardcoding (since enum cannot be used in constant, then maybe replace enum with constants… I know it seems pretty cumbersome).
Alternatively maybe we could just add some comment here or in entity _types.rs what is the `EntityType` in decimal.

By replacing enum with constant I mean something like this
```rust
pub mod EntityType {
pub const GlobalPackage: u32 = 0b00001101,
  …
}
```

Reply:
So this actually fails at compile time (due to the const functions) if there's an issue -- and I also assert that it's of the right type in the tests below.

I quite like having the address in this format because it's what the generation script outputs; and I quite like having the entity types in that format because it's easier to sanity-check against the Bech32.

I'm happy to add the decimal as a comment to EntityType though :)

Reply:
Sure, fair enough :) thanks!